src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/special/SpecialVideosPageObject.java

```diff
@@ -114,33 +104,47 @@ public void deleteVideo() {
 
   public void verifyDeleteViaGlobalNotifications() {
     YoutubeVideo video = YoutubeVideoProvider.getLatestVideoForQuery("truth");
-
+    System.out.println("Video: " + video.getTitle());
+    System.out.println("Get URL: " + video.getUrl());
     addVideoViaAjax(video.getUrl());
-    deleteVideo();
+    deleteNewestVideo();
     String deletedVideo = "\"File:" + video.getTitle() + "\" has been deleted. (undelete)";
+    System.out.println("BannerNotificationText: " + getBannerNotificationText());
+    System.out.println("Deleted video: " + deletedVideo);
     Assertion.assertEquals(getBannerNotificationText(), deletedVideo);
     PageObjectLogging.log("verifyDeleteVideoGlobalNotifications", "verify video " + deletedVideo
         + " was deleted", true);
   }
 
-  public void verifyDeleteViaVideoNotPresent() {
-    YoutubeVideo video = YoutubeVideoProvider.getLatestVideoForQuery("truth");
+  public boolean isHeaderVisible() {
+    try {
+      wait.forElementVisible(h1Header);
+      return true;
+    } catch (NoSuchElementException e) {
+      PageObjectLogging.log("verifyElementsOnPage", "verify that H1 is present", true);
+      return false;
+    }
+  }
 
-    addVideoViaAjax(video.getUrl());
-    deleteVideo();
-    getBannerNotifications().verifyNotificationMessage();
-    Assertion.assertNotEquals(getNewestVideoTitle(), video.getTitle());
-    PageObjectLogging.log("verifyDeleteVideoNotPresent", "verify video " + video.getTitle()
-        + " was deleted", true);
+  public boolean isAddVideoButtonClickable() {
+    try {
+      wait.forElementClickable(addVideo);
+      return true;
+    } catch (NoSuchElementException e) {
+      PageObjectLogging.log("verifyElementsOnPage", "verify that Add Video button is present", true);
+      return false;
+    }
   }
 
-  public void verifyElementsOnPage() {
-    verifyH1();
-    PageObjectLogging.log("verifyElementsOnPage", "verify that H1 is present", true);
-    verifyAddVideoButton();
-    PageObjectLogging.log("verifyElementsOnPage", "verify that Add Video button is present", true);
-    verifyNewestVideo();
-    PageObjectLogging.log("verifyElementsOnPage",
-        "verify that there is at least one video present", true);
+  public boolean isNewestVideoVisible() {
+    try {
+      wait.forElementVisible(newestVideo);
+      return true;
+    } catch (NoSuchElementException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)